### ISSUE TITLE
to achieve same results as presented in the paper

### ISSUE THREAD

**Issue by griffintin** [wvangansbeke/Sparse-Depth-Completion#19]
@wvangansbeke 

Thank you for sharing code.

I read though the code, but found that  "mse loss" is by default  used for training in given Shell/train.sh. At least, loss with uncertainty should be used, i guess. And Have no idea whether other options should be changed too.

Is it possible to share your training scripts which can achieve the same results as presented in the paper?

**Comment by wvangansbeke** [comment]
Hi @griffintin,

I increased the stability of the training process a while ago and I also made it converge faster by adding skip connections between the global and local network. Initially I only used guidance by multiplication with an attention map (=probability), but found out that it is less robust and that differences between a focal MSE and vanilla MSE loss function were now negligible. Be aware that this change will alter the appearance of the confidence maps since fusion takes place at mutliple stages now (see README). 

In short: if you want the old version, just remove the skip connections to get the original confidence maps.  The results should be similar and you should get a sense of interpretability. If only the numbers on the benchmark matter, don't alter it.

Best,
Wouter

**Comment by griffintin** [comment]
@wvangansbeke 

Thank you for quick reply, really appreciate it.

Yes, before I posted my questions , I have noticed your recent improvement with the skip connections between global-local networks.

Still, I am not clear about the loss functions.  Since both networks need to learn the uncertainty maps, I guess "MSE_loss_uncertainty" should be used instead of the default mse_loss.

Please correct me if I miss-understood.
 
Regards

**Comment by wvangansbeke** [comment]
Hi @wvangansbeke,

Yes you can but the differences will be small. Try with a vanilla MSE loss first to make sure the rest is correct (e.g. dataloading).

Best,
Wouter
